server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreRepositoryRegistrationIT.java

```diff
@@ -0,0 +1,103 @@
+
+        String settingsAttributeKeyPrefix = String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX, name);
+        Map<String, String> settingsMap = node.getAttributes()
+            .keySet()
+            .stream()
+            .filter(key -> key.startsWith(settingsAttributeKeyPrefix))
+            .collect(Collectors.toMap(key -> key.replace(settingsAttributeKeyPrefix, ""), key -> node.getAttributes().get(key)));
+
+        Settings.Builder settings = Settings.builder();
+        settingsMap.entrySet().forEach(entry -> settings.put(entry.getKey(), entry.getValue()));
+        settings.put(BlobStoreRepository.SYSTEM_REPOSITORY_SETTING.getKey(), true);
+
+        return new RepositoryMetadata(name, type, settings.build());
+    }
+
+    private void assertRemoteStoreRepositoryOnAllNodes() {
+        RepositoriesMetadata repositories = internalCluster().getInstance(ClusterService.class, internalCluster().getNodeNames()[0])
+            .state()
+            .metadata()
+            .custom(RepositoriesMetadata.TYPE);
+        RepositoryMetadata actualSegmentRepository = repositories.repository(REPOSITORY_NAME);
+        RepositoryMetadata actualTranslogRepository = repositories.repository(REPOSITORY_2_NAME);
+
+        for (String nodeName : internalCluster().getNodeNames()) {
+            ClusterService clusterService = internalCluster().getInstance(ClusterService.class, nodeName);
+            DiscoveryNode node = clusterService.localNode();
+            RepositoryMetadata expectedSegmentRepository = buildRepositoryMetadata(node, REPOSITORY_NAME);
+            RepositoryMetadata expectedTranslogRepository = buildRepositoryMetadata(node, REPOSITORY_2_NAME);
+            assertTrue(actualSegmentRepository.equalsIgnoreGenerations(expectedSegmentRepository));
+            assertTrue(actualTranslogRepository.equalsIgnoreGenerations(expectedTranslogRepository));
+        }
+    }
+
+    public void testSingleNodeClusterRepositoryRegistration() throws Exception {
+        internalCluster().startNode();
+        assertRemoteStoreRepositoryOnAllNodes();
+    }
+
+    public void testMultiNodeClusterRepositoryRegistration() throws Exception {
+        internalCluster().startNodes(3);
+        assertRemoteStoreRepositoryOnAllNodes();
+    }
+
+    public void testMultiNodeClusterRepositoryRegistrationWithMultipleMasters() throws Exception {
+        internalCluster().startClusterManagerOnlyNodes(3);
+        internalCluster().startNodes(3);
+        assertRemoteStoreRepositoryOnAllNodes();
+    }
+
+    public void testMultiNodeClusterActiveMasterShutDown() throws Exception {
+        internalCluster().startNodes(3);
+        internalCluster().stopCurrentClusterManagerNode();
+        ensureStableCluster(2);
+        assertRemoteStoreRepositoryOnAllNodes();
+    }
+
+    public void testMultiNodeClusterRandomNodeShutDown() throws Exception {
```

Comment:
Do we have a test for Network disruption where remote store nodes join -> drop -> re-join as the partition sets-in and heals